08-5-ex-sol-ADHD.Rmd

```diff
@@ -0,0 +1,166 @@
+rowdata_modified <- BiocParallel::bplapply(rowData(se), 
+                                           FUN = stringr::str_remove, 
+                                           pattern = '.*[kpcofg]__')
+# Genus level has additional '\"', so let's delete that also
+rowdata_modified <- BiocParallel::bplapply(rowdata_modified, 
+                                           FUN = stringr::str_remove, 
+                                           pattern = '\"')
+# rowdata_modified is a list, so it is converted back to DataFrame format. 
+rowdata_modified <- DataFrame(rowdata_modified)
+# And then assigned back to the SE object
+rowData(se) <- rowdata_modified
+# We use this to check what type of data it is
+# read.table(sample_meta_file_path)
+# It seems like a comma separated file and it does not include headers
+# Let us read it and then convert from data.frame to DataFrame
+# (required for our purposes)
+sample_meta <- DataFrame(read.table(sample_meta_file_path, sep = ",", header = FALSE))
+# Add sample names to rownames
+rownames(sample_meta) <- sample_meta[,1]
+# Delete column that included sample names
+sample_meta[,1] <- NULL
+# We can add headers
+colnames(sample_meta) <- c("patient_status", "cohort", "patient_status_vs_cohort", "sample_name")
+# Then it can be added to colData
+colData(se) <- sample_meta
+# Convert to tse format
+tse <- as(se, "TreeSummarizedExperiment")
+# Reads the tree file
+tree <- ape::read.tree(tree_file_path)
+# Add tree to rowTree
+rowTree(tse) <- tree
+```
+
+
+ * Visualize community variation with PCoA.
+ 
+```{r, message=FALSE, warning=FALSE}
+# Computing and storing relative abundance
+tse <- transformCounts(tse, method = "relabundance")
+
+# Gets relative bundance table
+relabundance_assay <- assays(tse)$relabundance
+
+# Transposes it to get taxa to columns
+relabundance_assay <- t(relabundance_assay)
+
+# Calculates Euclidean distances between samples.
+euclidean_dist_relA <- vegan::vegdist(relabundance_assay, method = "euclidean")
+
+# Performing PCoA
+euclidean_relA_pcoa <- ecodist::pco(euclidean_dist_relA)
+
+# Creates a data frame from principal coordinates
+euclidean_relA_pcoa_df <- data.frame(pcoa1 = euclidean_relA_pcoa$vectors[,1], 
+                                pcoa2 = euclidean_relA_pcoa$vectors[,2])
+
+# Visualizing
+library(ggplot2)
+euclidean_relA_plot <- ggplot(data = euclidean_relA_pcoa_df, aes(x=pcoa1, y=pcoa2)) +
+  geom_point() +
+  labs(x = "Principal Component 1",
+       y = "Principal Component 2",
+       title = "Euclidean PCoA with Relative Abundance transformation") +
+  theme(title = element_text(size = 12)) # makes titles smaller
+
+euclidean_relA_plot
+```
+
+ * Investigate the influence of the data transformations on
+   statistical analysis: Visualize community variation with PCoA with
+   the following options: 1) Bray-Curtis distances for compositional
+   data; 2) Euclidean distances for CLR-transformed data.
+
+```{r}
+ # Computing Bray-Curtis distances
```

Comment:
BC is not (mathematical) distance. It is dissimilarity. We like to enforce the use of correct terminology, change this.